kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -360,6 +380,19 @@ public String marshall(Object objectInput) {
         }
     }
 
+    @Override
+    public <T> T unmarshall(byte[] serializedInput, Class<T> type) {
+
+        try {
+            Class<?> actualType = classesSet.contains(type) ? Object.class : type;
```

Comment:
hmmmm, not sure about this one, this was copy paste of existing unmarshall, changing that might have unexpected effects. 
I will open a different JIRA, since this seems a bug to me